src/registrar/views/application.py

```diff
@@ -481,5 +483,24 @@ class ApplicationStatus(generic.DetailView):
     template_name = "application_status.html"
 
     def get_context_data(self, **kwargs):
+        """Get context details to process information from application"""
         context = super(ApplicationStatus, self).get_context_data(**kwargs)
         return context
+
+
+class ApplicationWithdraw(generic.DetailView, DomainPermission):
+    model = DomainApplication
+    template_name = "application_withdraw_confirmation.html"
+    """ The page above will display asking user to confirm if they want to withdraw;
+
```

Comment:
@kewlguy781 I went to retest this for the login issue. I cannot withdraw as a logged out user 👏 but I am able to see the content on both https://getgov-jon.app.cloud.gov/application/3 and https://getgov-jon.app.cloud.gov/application/3/withdraw. This might be a bug in `DomainPermission` itself, or it might be that additional mixins are needed.

Reply:
this ticket focus only on the withdraw, the ticket for the application part is already merged into main. I will make change under different branch.